docker-entrypoint.sh

```diff
@@ -1,61 +1,366 @@
+  "WARN")
+    container_log_level_int=30
+    ;;
+  "ERROR")
+    container_log_level_int=40
+    ;;
+  "CRITICAL")
+    container_log_level_int=50
+    ;;
+  *)
+    debug "No matching log level found: ${CONTAINER_LOG_LEVEL}."
+    debug "Defaulting to INFO."
+    CONTAINER_LOG_LEVEL="INFO"
+    container_log_level_int=20
+    ;;
+  esac
+  success "CONTAINER_LOG_LEVEL: ${CONTAINER_LOG_LEVEL}. Set CONTAINER_LOG_LEVEL=DEBUG for more details."
+}
+
+update_file() {
+  local target_file_path="$1"
+  local original_file_path="$2"
+  # Default to false
+  local print_migration_info_on_update="${3:-false}"
+
+  if [ ! -f "${target_file_path}" ] || ! cmp -s "${original_file_path}" "${target_file_path}"; then
+    success "Update the file ${target_file_path} with ${original_file_path}"
+    cp -f "${original_file_path}" "${target_file_path}" || warning "Could not copy ${original_file_path} to ${target_file_path}"
+    if [ "${print_migration_info_on_update}" = "true" ]; then
+      print_migration_info="true"
+    fi
+  else
+    success "The file ${target_file_path} is up to date"
+  fi
+}
+
+extract_config_info() {
+  local config_location="$1"
+  local config_variable="$2"
+  local config_value=""
+  if [[ "${config_location}" = *.yml ]]; then
+    config_value=$(yq -r "${config_variable}" "${config_location}")
+  elif [[ "${config_location}" = *.json ]]; then
+    config_value=$(jq -r "${config_variable}" "${config_location}")
+  fi
+  # Validate the config value
+  if [ -z "${config_value}" ] || [ "${config_value}" = null ]; then
+    config_value=""
+  fi
+  # Return the value
+  echo "${config_value}"
+}
+
+echo "#################"
+echo "# Container ENV #"
+echo "#################"
+# Validate the log level
+set_log_level
+
+# Set the jar file location
+jar_file=/ors.jar
+BUILD_GRAPHS=${BUILD_GRAPHS:-"false"}
+REBUILD_GRAPHS=${REBUILD_GRAPHS:-"false"}
+# If BUILD_GRAPHS is set to true, we need to set ors_rebuild_graphs to true and print an info about migration to REBUILD_GRAPHS
+if [ "${BUILD_GRAPHS}" = "true" ]; then
+  ors_rebuild_graphs="true"
+  warning "BUILD_GRAPHS is deprecated and will be removed in the future."
+  warning "Please use REBUILD_GRAPHS instead."
+elif [ "${REBUILD_GRAPHS}" = "true" ]; then
+  ors_rebuild_graphs="true"
+else
+  ors_rebuild_graphs="false"
+fi
+
+
+# get ors.engine.graphs_root_path=. Dot notations in bash are not allowed, so we need to use awk to parse it.
+ors_engine_graphs_root_path=$(env | grep "^ors\.engine\.graphs_root_path=" | awk -F '=' '{print $2}')
+# get ors.engine.elevation.cache_path
+ors_engine_elevation_cache_path=$(env | grep "^ors\.engine\.elevation\.cache_path=" | awk -F '=' '{print $2}')
+# get ors.engine.source_file
+ors_engine_source_file=$(env | grep "^ors\.engine\.source_file=" | awk -F '=' '{print $2}')
+# get ors_config_location
+ors_config_location=${ORS_CONFIG_LOCATION:-""}
+# Unset the ORS_CONFIG_LOCATION to not interfere with the spring-boot application
+unset ORS_CONFIG_LOCATION
```

Comment:
This is only used in line 217, so we should move it there?

Reply:
Yes, but since these variables are only used there and possibly even overwritten, I'd say the belong to that section, and shouldn't be declared way before, so no one remembers what they even were?

Reply:
Well, it's a matter of taste. Go ahead and move them if you think it brings more value :+1: